ibis/common/tests/test_patterns.py

```diff
@@ -495,6 +554,60 @@ def test_matching():
     assert match(InstanceOf(object) & InstanceOf(float), 3.14) == 3.14
 
 
+def test_replace_passes_matched_value_as_underscore():
+    class MyInt:
+        def __init__(self, value):
+            self.value = value
+
+        def __eq__(self, other):
+            return self.value == other.value
+
+    p = InstanceOf(int) >> Call(MyInt, value=_)
+    assert p.match(1, context={}) == MyInt(1)
+
+
+def test_replace_in_nested_object_pattern():
+    # simple example using reference to replace a value
+    b = Variable("b")
+    p = Object(Foo, 1, b=b)
+    f = p.match(Foo(1, 2), {b: 3})
+    assert f.a == 1
+    assert f.b == 3
+
+    # nested example using reference to replace a value
+    d = Variable("d")
+    p = Object(Foo, 1, b=Object(Bar, 2, d=d))
+    g = p.match(Foo(1, Bar(2, 3)), {d: 4})
+    assert g.b.c == 2
+    assert g.b.d == 4
+
+    # nested example using reference to replace a value with a captured value
+    p = Object(
+        Foo,
+        1,
+        b=Replace(
+            Object(Bar, 2, d="d" @ Any()), lambda value, ctx: Foo(-1, b=ctx["d"])
+        ),
+    )
```

Comment:
🤯

Reply:
Yeah, this is not the nicest way to spell it. There is already some syntax sugar which improves it, but I'm definitely open for suggestions!